great_expectations/expectations/metrics/map_metric_provider/map_metric_provider.py

```diff
@@ -628,25 +622,4 @@ def _get_evaluation_dependencies(
                     metric_value_kwargs=base_metric_value_kwargs,
                 )
 
-        return dependencies
-
-    @staticmethod
-    def is_sqlalchemy_metric_selectable(
-        map_metric_provider: MetaMetricProvider,
-    ) -> bool:
-        """
-        :param map_metric_provider: object of type "MapMetricProvider", whose SQLAlchemy implementation is inspected
-        :return: boolean indicating whether or not the returned value of a method implementing the metric resolves all
-        columns -- hence the caller must not use "select_from" clause as part of its own SQLAlchemy query; otherwise an
-        unwanted selectable (e.g., table) will be added to "FROM", leading to duplicated and/or erroneous results.
-        """
-        # noinspection PyUnresolvedReferences
-        return (
-            hasattr(map_metric_provider, "condition_metric_name")
-            and map_metric_provider.condition_metric_name
-            in MapMetricProvider.SQLALCHEMY_SELECTABLE_METRICS
-        ) or (
-            hasattr(map_metric_provider, "function_metric_name")
-            and map_metric_provider.function_metric_name
-            in MapMetricProvider.SQLALCHEMY_SELECTABLE_METRICS
-        )
+        return dependencies
```

Comment:
I don't think there's any harm in moving this method outside the class, since it's a static method.

However, this is technically a breaking change. 🤔